### DISCUSSION TITLE
[9 gennaio 2023] Automi

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#1] [Automi]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/5b5721a9-add6-4c8c-8e4f-8e5bc082fc21)

**Comment by Exyss** [comment]
Per il primo punto, ripropongo la soluzione presente sui miei appunti:

![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/0ef44fab-c577-4737-8262-ae006b681d3b)
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/58d2a537-e0ae-48a7-a21a-668c0f167b74)

-----

La dimostrazione richiesta nel secondo punto è presente sul libro di testo

**Reply by MrPanas** [reply]
Sono giunto alla stessa conclusione.

**Reply by Ivanettoss** [reply]
attenzione typo nella prima versione, è b(b U c) e non b(a U c)

**Comment by daromacarlo** [comment]
<img width="1616" height="692" alt="Immagine 2026-02-02 165915" src="https://github.com/user-attachments/assets/2fab00a0-7021-4930-94bb-b77a834a80bd" />
<img width="1617" height="841" alt="Immagine 2026-02-02 165928" src="https://github.com/user-attachments/assets/b3cc8985-c77a-4584-a7c3-3d6c89ff9967" />
<img width="1619" height="248" alt="Immagine 2026-02-02 165942" src="https://github.com/user-attachments/assets/98045a50-ad65-4ecc-a87c-87f7517f326b" />
<img width="1620" height="770" alt="Immagine 2026-02-02 165959" src="https://github.com/user-attachments/assets/a8b99bc9-ef04-4b89-b74f-6ef671f62ad9" />
<img width="1621" height="656" alt="Immagine 2026-02-02 170023" src="https://github.com/user-attachments/assets/a2c03543-bbe6-4312-86f0-d01d449a957d" />
<img width="1620" height="267" alt="Immagine 2026-02-02 170057" src="https://github.com/user-attachments/assets/651307f6-1ed7-426d-89d6-d6824dcaf4a9" />

**Comment by Pierba** [comment]
Nel secondo punto dell'esercizio mi sento di dire che le regole nella forma A $\rightarrow$ BC sono composte da A $\in$ V e B, C $\in$ V \ {S}. Nè B nè C possono essere terminali proprio perchè c'è come passo dell'algoritmo quello della sostituzione di tutte le regole A $\rightarrow$ u1u2 dove u1 o u2 possono essere sia variabili che terminali, perciò se sono terminali vengono aggiunte le regole U1 $\rightarrow$ u1 e/o U2 $\rightarrow$ u2 a seconda dei casi. Per concludere le regole nella forma normale di Chomsky si presentano solo nella forma di A $\rightarrow$ BC (con B e C entrambe variabili, eslusa iniziale S), A $\rightarrow$ a (con a terminale) e S $\rightarrow \epsilon$   (l'unica $\epsilon$-regola ammessa).
